babel/dates.py

```diff
@@ -1282,6 +1284,16 @@ def format_timezone(self, char, num):
                 return get_timezone_name(self.value.tzinfo, width,
                                          uncommon=True, locale=self.locale)
             return get_timezone_location(self.value.tzinfo, locale=self.locale)
+        # Included additional elif condition to add support for 'X' in timezone format
+        elif char in "xX":
+            if get_timezone_gmt(self.value, width='short', locale=self.locale)=='+0000' and char == 'X':
+                return 'Z'
+            elif num == 1:
+                return get_timezone_gmt(self.value, width='short', locale=self.locale)[:-2]
```

Comment:
This will return incorrect values for timezones that are not an integer number of hours away from GMT.

The TR-35 document states that the minutes part is _optional_, i.e. if it were `00`, it will be omitted, not that it's always truncated out.